pkg/storegateway/series_refs_test.go

```diff
@@ -580,6 +580,90 @@ func TestMergedSeriesChunkRefsSet(t *testing.T) {
 	}
 }
 
+func BenchmarkMergedSeriesChunkRefsSetIterators(b *testing.B) {
+	const (
+		numSetsPerIterator = 10
+		numSeriesPerSet    = 10
+		mergedBatchSize    = 5000
+	)
+
+	// Creates the series sets, guaranteeing series sorted by labels.
+	createUnreleasableSets := func(iteratorIdx int) []seriesChunkRefsSet {
+		sets := make([]seriesChunkRefsSet, 0, numSetsPerIterator)
+		for setIdx := 0; setIdx < numSetsPerIterator; setIdx++ {
+			// This set cannot be released because reused between multiple benchmark runs.
+			set := newSeriesChunkRefsSet(numSeriesPerSet, false)
+
+			for seriesIdx := 0; seriesIdx < numSeriesPerSet; seriesIdx++ {
+				set.series = append(set.series, seriesChunkRefs{
+					lset: labels.New(labels.Label{Name: labels.MetricName, Value: fmt.Sprintf("metric_%06d", (iteratorIdx*numSetsPerIterator*numSeriesPerSet)+(setIdx*numSeriesPerSet)+seriesIdx)}),
+				})
+			}
+
+			sets = append(sets, set)
+		}
+
+		return sets
+	}
+
+	for _, withDuplicatedSeries := range []bool{true, false} {
+		for numIterators := 1; numIterators < 100; numIterators = numIterators * 2 {
```

Comment:
```suggestion
		for numIterators := 1; numIterators <= 64; numIterators *= 2 {
```

Reply:
Fixed in https://github.com/grafana/mimir/pull/3666/commits/94677528779acd5446d4419bcb72eef4e9033e2a